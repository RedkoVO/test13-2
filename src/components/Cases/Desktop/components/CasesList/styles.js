import Casestar from '../../../../../assets/images/svg/Casestar.svg'

export default () => ({
  root: {
    display: 'flex',
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'space-between',
    flexWrap: 'wrap',
    width: '800px',
    marginTop: '25px'
  },
  casesListItem: {
    position: 'relative',
    display: 'flex',
    flexDirection: 'column',
    alignItems: 'center',
    width: '170px',
    height: '200px',
    backgroundColor: 'rgba(0,0,0,0.15)',
    borderRadius: '15px',
    justifyContent: 'space-around',
    boxSizing: 'border-box',
    padding: '10px 0',
    marginBottom: '25px'
  },
  casesListItemImg: {
    width: '150px',
    height: '125px',

    '& img': {
      maxWidth: '100%'
    }
  },
  casesListItemName: {
    fontSize: '13px',
    marginBottom: '5px',

    '& span': {
      fontWeight: 500
    }
  },
  casesListItemPrise: {
    fontSize: '10px',
    padding: '5px 15px',
    boxSizing: 'border-box',
    borderRadius: '10px',
    backgroundColor: 'rgba(0,0,0,0.13)',

    '& span': {
      fontWeight: 500
    }
  },
  casesListItemGame: {
    fontSize: '9px',

    '& span': {
      fontWeight: 500
    }
  },
  casesListItemMarker: {
    position: 'absolute',
    width: '13px',
    height: '13px',
    background: `url(${Casestar})`,
    backgroundRepeat: 'no-repeat',
    backgroundPosition: 'center',
    backgroundSize: 'cover',
    right: '10px',
    bottom: '10px'
  }
})
