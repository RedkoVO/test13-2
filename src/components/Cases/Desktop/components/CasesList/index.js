import React from 'react'
import PropTypes from 'prop-types'
import withStyles from '@material-ui/core/styles/withStyles'

import styles from './styles'

const CasesList = ({ classes, cases }) => (
  <div className={classes.root}>
    {cases.map(item => (
      <div className={classes.casesListItem} key={item.id}>
        <div className={classes.casesListItemImg}>
          <img src={item.img} alt="" />
        </div>
        <div className={classes.casesListItemName}>
          Case: <span>{item.title}</span>
        </div>
        <div className={classes.casesListItemPrise}>
          Open for <span>{item.cost}</span>
        </div>
        <div className={classes.casesListItemGame}>
          Case: <span>{item.game}</span>
        </div>
        <div className={classes.casesListItemMarker} />
      </div>
    ))}
  </div>
)

CasesList.propTypes = {
  classes: PropTypes.object,
  cases: PropTypes.array
}

export default withStyles(styles)(CasesList)
