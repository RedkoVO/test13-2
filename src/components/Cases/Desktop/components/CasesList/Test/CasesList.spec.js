import * as React from 'react'
import renderer from 'react-test-renderer'

import CasesList from '../'

import CaseXIcon from '../../../../../../assets/images/caseX.png'

const tmpCases = [
  {
    id: 1,
    img: CaseXIcon,
    title: 'Hello, buddy!',
    cost: '$34.20',
    game: 'CS:GO'
  },
  {
    id: 2,
    img: CaseXIcon,
    title: 'Hello, buddy!',
    cost: '$34.20',
    game: 'CS:GO'
  },
  {
    id: 3,
    img: CaseXIcon,
    title: 'Hello, buddy!',
    cost: '$34.20',
    game: 'CS:GO'
  },
  {
    id: 4,
    img: CaseXIcon,
    title: 'Hello, buddy!',
    cost: '$34.20',
    game: 'CS:GO'
  }
]

it('renders correctly', () => {
  const tree = renderer.create(<CasesList cases={tmpCases} />).toJSON()
  expect(tree).toMatchSnapshot()
})
