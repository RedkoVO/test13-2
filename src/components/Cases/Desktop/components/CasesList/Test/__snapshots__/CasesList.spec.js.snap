// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="CasesList-root-1"
>
  <div
    className="CasesList-casesListItem-2"
  >
    <div
      className="CasesList-casesListItemImg-3"
    >
      <img
        alt=""
        src="caseX.png"
      />
    </div>
    <div
      className="CasesList-casesListItemName-4"
    >
      Case: 
      <span>
        Hello, buddy!
      </span>
    </div>
    <div
      className="CasesList-casesListItemPrise-5"
    >
      Open for 
      <span>
        $34.20
      </span>
    </div>
    <div
      className="CasesList-casesListItemGame-6"
    >
      Case: 
      <span>
        CS:GO
      </span>
    </div>
    <div
      className="CasesList-casesListItemMarker-7"
    />
  </div>
  <div
    className="CasesList-casesListItem-2"
  >
    <div
      className="CasesList-casesListItemImg-3"
    >
      <img
        alt=""
        src="caseX.png"
      />
    </div>
    <div
      className="CasesList-casesListItemName-4"
    >
      Case: 
      <span>
        Hello, buddy!
      </span>
    </div>
    <div
      className="CasesList-casesListItemPrise-5"
    >
      Open for 
      <span>
        $34.20
      </span>
    </div>
    <div
      className="CasesList-casesListItemGame-6"
    >
      Case: 
      <span>
        CS:GO
      </span>
    </div>
    <div
      className="CasesList-casesListItemMarker-7"
    />
  </div>
  <div
    className="CasesList-casesListItem-2"
  >
    <div
      className="CasesList-casesListItemImg-3"
    >
      <img
        alt=""
        src="caseX.png"
      />
    </div>
    <div
      className="CasesList-casesListItemName-4"
    >
      Case: 
      <span>
        Hello, buddy!
      </span>
    </div>
    <div
      className="CasesList-casesListItemPrise-5"
    >
      Open for 
      <span>
        $34.20
      </span>
    </div>
    <div
      className="CasesList-casesListItemGame-6"
    >
      Case: 
      <span>
        CS:GO
      </span>
    </div>
    <div
      className="CasesList-casesListItemMarker-7"
    />
  </div>
  <div
    className="CasesList-casesListItem-2"
  >
    <div
      className="CasesList-casesListItemImg-3"
    >
      <img
        alt=""
        src="caseX.png"
      />
    </div>
    <div
      className="CasesList-casesListItemName-4"
    >
      Case: 
      <span>
        Hello, buddy!
      </span>
    </div>
    <div
      className="CasesList-casesListItemPrise-5"
    >
      Open for 
      <span>
        $34.20
      </span>
    </div>
    <div
      className="CasesList-casesListItemGame-6"
    >
      Case: 
      <span>
        CS:GO
      </span>
    </div>
    <div
      className="CasesList-casesListItemMarker-7"
    />
  </div>
</div>
`;
